tests/functional/test_paramfile.py

```diff
@@ -0,0 +1,186 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import json
+import logging
+
+from mock import patch, Mock, mock_open
+
+from awscli.testutils import FileCreator, BaseCLIWireResponseTest
+from awscli.clidriver import create_clidriver
+from botocore.vendored import requests
+
+logger = logging.getLogger(__name__)
+
+
+class FakeResponse(object):
+    def __init__(self, *args, **kwargs):
+        self.status_code = 200
+        self.text = 'function-name'
+
+
+class BaseTestCLIFollowParamURL(BaseCLIWireResponseTest):
+    def setUp(self):
+        super(BaseTestCLIFollowParamURL, self).setUp()
+        self.wire_response = json.dumps({}).encode('utf-8')
+        self.patch_send(content=self.wire_response)
+        self.files = FileCreator()
+        self.command_prefix = ['lambda', 'get-function', '--function-name']
+        self.start_patches()
+
+    def tearDown(self):
+        super(BaseTestCLIFollowParamURL, self).tearDown()
+        self.files.remove_all()
+        self.stop_patches()
+
+    def start_patches(self):
```

Comment:
I prefer we avoid the pattern in general of starting the patches in the `setUp` and tearing them down in the `tearDown()`. Instead, let's either try to avoid patching at all cost for functional tests or use context managers. When they are in the `setUp` and `tearDown` it is pretty easy to forget to tear down a patch causing issues down the road when more tests get added.